tests/tape/test_tape.py

```diff
@@ -343,21 +345,92 @@ def make_extendible_tape(self):
 
         return tape
 
+    def test_specs_empty_tape(self, make_empty_tape):
+        """Test specs attribute on an empty tape"""
+        tape = make_empty_tape
+
+        assert tape.specs["by_size"] == defaultdict(int)
+        assert tape.specs["by_name"] == defaultdict(int)
+
+        assert tape.specs["total_operations"] == 0
+        assert tape.specs["total_observables"] == 1
+        assert tape.specs["num_tape_wires"] == 2
+        assert tape.specs["num_trainable_params"] == 0
+        assert tape.specs["depth"] == 0
+
+        assert len(tape.specs) == 7
```

Comment:
Oh nice, ignore my comment above then!

Reply:
As I found out, empty circuits have their own peculiarties, so I'm adding tests for empty qnodes.